File: Data Structures & Algorithms/top-k-elements-in-list/submission-3.kt
class Solution {
    fun topKFrequent(nums: IntArray, k: Int): IntArray {
        val hashMap = hashMapOf<Int,Int>()
        val rank = hashMapOf<Int,Int>()
        val result = mutableListOf<Int>()
        for(n in nums){
            hashMap[n] = hashMap.getOrDefault(n, 0) + 1
        }
        val sortedMap = hashMap.toList()
            .sortedByDescending { it.second }
            .toMap()
        var found = 0
        for((key,value) in sortedMap){
            if(result.isNotEmpty() && result[result.size - 1] != key){
                found++
            }
            if(found < k){
                result.add(key)
            }
        }
        return result.toIntArray()
    }
}

File: Data Structures & Algorithms/climbing-stairs/submission-3.cpp
class Solution {
public:
    int climbStairs(int n) {
        int n0 = 0;
        int n1 = 1;
        int result;
        for(int i = 0; i < n; i++){
            result = n0 + n1;
            n0 = n1;
            n1 = result;
        }
        return result;
    }
};
